### ISSUE TITLE
Don't use the wording "quit unpextedly" for python exceptions

### ISSUE THREAD

**Issue by elad661** [abrt/gnome-abrt#86]
Some python apps spew exception stacktraces but don't quit. gnome-abrt is an example of such app (just search for something long enough that will have no results in your issue list).

This kind of exception should be classified as "misbehavior" unless they caused the program to actually quit. Saying "quit unexpectedly" when the app never quit is confusing.

If we don't have a reliable way to tell if a Python app crashed because of an exception or kept going, we should probably change the wording from "quit unexpectedly" to "encountered a problem" for Python apps, because we can't be sure if they quit or not.

**Comment by xsuchy** [comment]
If it quit unexpectedly and then respawn, it is still quit. I do not think we need change the wording.

**Comment by elad661** [comment]
>If it quit unexpectedly and then respawn, it is still quit. I do not think we need change the wording.

Some python apps survive exceptions *without* quitting. It's not "quit and respawn", the app keeps working and doesn't close. They basically have a catch-all handler that prints the stacktrace, to avoid the program quitting on an unhandeled exception.

**Comment by xsuchy** [comment]
But afaik abrt does not report those applications. Or do you have reproducer?

**Comment by jfilak** [comment]
@xsuchy Have you ever tried to develop a python program using threads on GLib main loop?

**Comment by jfilak** [comment]
> But afaik abrt does not report those applications. Or do you have reproducer?

What about the reproducer in the first comment? Is that enough?

**Comment by xsuchy** [comment]
Well unhandled crash in thread is still "quit unexpectedly". And handling the case in another process by respawning the the thread is very very bad design.

**Comment by jfilak** [comment]
> Well unhandled crash in thread is still "quit unexpectedly".

Yes, the thread quit unexpectedly but the application is still running, hence users get confused. In case of gnome-abrt, there isn't even an extra thread. The exception handler is somehow invoked by the Python bindings.

> And handling the case in another process by respawning the the thread is very very bad design.

What are you talking about?

**Comment by xsuchy** [comment]
Yes, for investigation.

**Comment by ernestask** [comment]
Last I recall, Benjamin Berg worked on GNOME sessions as systemd user sessions and it should be in a usable state these days. Not sure exactly how that helps, but I’ll see if we can detect these cases somehow.
